dev/RadioButtons/RadioButtons.h

```diff
@@ -53,8 +52,7 @@ class RadioButtons :
     void OnRepeaterElementClearing(const winrt::ItemsRepeater&, const winrt::ItemsRepeaterElementClearingEventArgs& args);
     void OnRepeaterElementIndexChanged(const winrt::ItemsRepeater&, const winrt::ItemsRepeaterElementIndexChangedEventArgs& args);
     void OnRepeaterCollectionChanged(const winrt::IInspectable&, const winrt::IInspectable&);
-    void OnChildChecked(const winrt::IInspectable& sender, const winrt::RoutedEventArgs& args);
-    void OnChildUnchecked(const winrt::IInspectable& sender, const winrt::RoutedEventArgs& args);
+    void OnChildClick(const winrt::IInspectable& sender, const winrt::RoutedEventArgs&);
```

Comment:
I don't kow what kind of technical reason it would be seeing as everything appears working as expected here. Then again, perhaps @StephenLPeters might know more who I believe did work on this control before.

Reply:
From a theoretical standpoint, RadioButtons shouldn't care if one of its radio button elements is clicked. What if there was a style change or a derived radio button class which didn't select when it was clicked.  What RadioButtons actually cares about is when one of its children becomes checked or unchecked, which is why we subscribed to these events.  With this change, what happens if you name one of the RadioButton elements and then check it programmatically, do we still fire a selection changed event?

Reply:
Also does this change break moving focus to a different radio button by using control key + arrow keys (to avoid selection follows focus) and then selecting by pressing the space button?  Looking at the change I don't see how this could support anything but selection by selection follows focus and clicking of radio buttons.

Reply:
@StephenLPeters 

> Also does this change break moving focus to a different radio button by using control key + arrow keys (to avoid selection follows focus) and then selecting by pressing the space button?

No, as the Click event is raised in this case and we update the selection as expected (including raising the event).

> With this change, what happens if you name one of the RadioButton elements and then check it programmatically, do we still fire a selection changed event?

Unfortunately, you raise a good point. We no longer notice this inside the RadioButtons control as part of this change.

This could make things very tricky, if not impossible here for us. If we want to raise the SelectionChanged event only *once* in every possible case selection changes for the RadioButtons control. then we cannot just listen to the Unchecked/Checked events of the RadioButton items. As these are two separate events, we don't know if a rasied Uncheck event will be followed by a rasied Checked event, so there is no way for us raise SelectionChanged correctly only *once*.

I'm not sure right now if we can find a solution for this...

Reply:
This is only annoying because RadioButton has logic in it that groups them together and maintains single selections within groups. If we could disable that behavior and manage single selection from the RadioButtons control then we could build the selection changed event args correctly.  We could either wait for Winui 3 where we can change RadioButton or we can try and find some creative way of disabling it, either by giving all of the radio button elements a unique group name or by trying to manipulate the default grouping behavior.  I beleive the default behavior is that all radio buttons with the same parent are put in the same group. So maybe we could have the factory put the radio buttons inside of a grid to block the grouping.

Reply:
> So maybe we could have the factory put the radio buttons inside of a grid to block the grouping.

I will give this a try.

Reply:
So...while I still cannot debug WinUI when running interaction tests, I managed to locate the error and got all RadioButtons tests to succeed. @StephenLPeters I followed your suggestion of wrapping the RadioButton control in a dedicated Grid (happening in `RadioButtonsElementFactory::GetElementCore()`). So now, each repeater element is by default a Grid and no longer a RadioButton. This does the job here.

However, I would like to first check in the upcoming PR for issue https://github.com/microsoft/microsoft-ui-xaml/issues/3159 as that will make the code for this PR easier then (especially now that we wrap each RadioButton in a Grid which will then be the actual RepeaterElement). A PR for issue #3159 will make sure that each RepeaterElement we get in RadioButtons is build the same way - that's not the case today. A developer can, today, specify a custom ItemTemplate which will then be the repeater element. No RadioButton wrapping involved. The repeater element will be the UIElement which is the root of the custom ItemTemplate.

Thus, we currently need to have checks in place to check if the RepeaterElement is a RadioButton (or Control) in order to call APIs like Checked/Unchecked/Focus()/.... A PR for issue #3159 will help simplify the code here as each repeater element then will always be a RadioButton, no matter if the developer supplied a custom ItemTemplate or not. I wouldn't have to focus on the cases any longer where the repeater element is not a grid wrapped around a RadioButton in case the developer supplied an own ItemTemplate.

Reply:
Sorry, I was on vacation last week, working my way through these now.  I was also worried about these changes happening in parallel and think that the order your proposed makes sense.

@ranjeshj the proposed solution to this but means that every radio button in a RadioButtons is going to have an extra grid associated with it.  I can't think of a better way to do this without WinUI 3 but I wonder if this is an instance of the cure being worse than the disease?

Reply:
@Felix-Dev are you still having debugging issue?

Reply:
No more debugging issues currently so at least on that front I could move ahead with this PR once #3194 has been merged (assuming it will at some point).

Reply:
Update: I can now debug interaction tests again. Not sure why it broke in the first place place but here's hoping it will keep working as it should.

I guess this conversation can now be marked as #resolved.